### ISSUE TITLE
BBC not downloading

### ISSUE THREAD

**Issue by forumite** [Ascoware/get-iplayer-automator#183]
ITV  is still fine.  
---------------

*What steps will reproduce the problem? Be specific, and provide as much detail as possible.*
Any BBC programme

*What is the expected output? What do you see instead?*

*What version of GiA are you using? On what version of OS X?*

*Are you using a web proxy? DNS proxy? VPN? VPS?*
DNS
*What program are you attempting to download (full title, series and episode)? **Provide only ONE program as an example.** *

*Please provide any additional information below.*
Downloading Show 1/1:

BBC Download (ID=b0b624f9): Downloading A Very English Scandal: Series 1 - Episode 3
WARNING: Ignoring invalid option(s) in /Users/Matthew/Library/Application Support/Get iPlayer Automator/options:
WARNING: rtmptvopts = --swfVfy http://www.bbc.co.uk/emp/releases/iplayer/revisions/617463_618125_4/617463_618125_4_emp.sw
WARNING: Please remove invalid options from /Users/Matthew/Library/Application Support/Get iPlayer Automator/options
INFO: Use --dump-options to display all valid options
INFO: 1 total programmes
WARNING: A UK TV licence is required to access BBC iPlayer TV content legally
INFO: Downloading tv: 'A Very English Scandal: Series 1 - 3. Episode 3 (b0b624f9) [original]'
INFO: Downloaded: 0.00 MB (00:00:00) @ 0.00 Mb/s (hvfhd1/bi) [audio+video]
WARNING: Unexpected size for file segment [1]
WARNING: Expected: 5231288  Downloaded: 1113044
WARNING: Retrying download

**Comment by darfielddx** [comment]
I have upgraded to 1.13.6 in the hope that I might get pictures that don't pixelate and lag. However, as with forumite above, I too cannot now download the BBC. 

My message said:
Downloading Show 1/2:

BBC Download (ID=b0b5y4zg): Downloading Suffragettes with Lucy Worsley - Suffragettes with Lucy Worsley
INFO: 1 total programmes
WARNING: A UK TV licence is required to access BBC iPlayer TV content legally
INFO: Downloading tv: 'Suffragettes with Lucy Worsley - - (b0b5y4zg) [original]'
INFO: Downloaded: 0.00 MB (00:00:00) @ 0.00 Mb/s (hvfxsd1/bi) [audio+video]
WARNING: Unexpected size for file segment [1]
WARNING: Expected: 1663800  Downloaded: 326616
WARNING: Retrying download
WARNING: Retrying tv: 'Suffragettes with Lucy Worsley - - (b0b5y4zg) [original]'

and so it went on for both the shows I was trying to download.

**Comment by skovatch** [comment]
Sounds like BBC changed something on their end that get_iplayer will need to handle. I am at a polling place all day today and won’t be able to look at anything until tomorrow.

**Comment by Vangelis66** [comment]
@skovatch wrote: 

> Sounds like BBC changed something on their end that get_iplayer will need to handle

Hello 😃 

All is relatively quite over at both the get_iplayer [support forum](https://forums.squarepenguin.co.uk/) and [mailing list](https://www.mail-archive.com/get_iplayer@lists.infradead.org/maillist.html) - hot topic is, unsurprisingly, the Dr Who (iPlayer-only) box set; had there been a major breakage caused by the beeb, those two would've been inundated with relevant reports... 😉

@forumite posted: 

> INFO: Downloaded: 0.00 MB (00:00:00) @ 0.00 Mb/s (hvfhd1/bi) [audio+video]
> WARNING: Unexpected size for file segment [1]
> WARNING: Expected: 5231288 Downloaded: 1113044

& @darfielddx posted: 

> INFO: Downloaded: 0.00 MB (00:00:00) @ 0.00 Mb/s (hvfxsd1/bi) [audio+video]
> WARNING: Unexpected size for file segment [1]
> WARNING: Expected: 1663800 Downloaded: 326616

Both download attempts involved the "**bidi**" CDN (tvmode=hvfhd1, CDN=bi), 
so perhaps there's something gone awry in that CDN (???)
There's a recent [addition](https://github.com/get-iplayer/get_iplayer/wiki/faq#what-does-it-mean-when-i-see-a-message-similar-to-warning-unexpected-size-for-file-segment-99) to the GiP wiki: 

> **What does it mean when I see a message similar to "WARNING: Unexpected size for file segment [99]"?**
> 
> This means a segment of the requested programme was incomplete. get_iplayer will make several 
> attempts to re-download the segment. This may be due to server problems or upstream connectivity 
> problems - it is not a get_iplayer problem. These problems are usually transient, so delete any partial 
> download and retry your download at a later time. 

which I think _may_ apply here... 
Perhaps explicitly request the same quality variant (hvfhd) from a different CDN (akamai | limelight), 
is this somehow possible in GiA? 
FWIW, I had no issues fetching --pid=b0b3lnks, --tvmode=hvfxsd3 (CDN=ak) with the standalone perl script 3.14 on Windows; so GiP isn't _broken_ per se: 
```
INFO: Downloading tv: 'The Sky at Night: Guides - 2. Galaxies (b0b3lnks) [original]'
  0.3% of ~887.40 MB @ ETA:  (hvfxsd3/ak) [audio+video]) 
```
Hope I've helped 😺

**Comment by skovatch** [comment]
@Vangelis66, thanks, that was indeed helpful. That FAQ does sound like the issue at hand. If you choose 'Best' as your BBC setting and turn off 'Force get_iplayer to ignore DASH video", 'tvbest' is used, which expands to "hvfhd,dvfhd,tvbetter" With those settings I'm able to download this as well.

**Comment by jchhenderson** [comment]
@skovatch ...did you mean check the box for 'Force get_iplayer to ignore DASH video" because I can download Dr. Who s01e01 with that setting checked and both "best" and "better" selected (best first).

**Comment by skovatch** [comment]
@jchhenderson I would leave that box unchecked. get_iplayer has fixed the problems with the audio encoding in DASH streams for iTunes, so there's no reason to ignore DASH anymore. In fact, I can probably remove that checkbox altogether, so get_iplayer can pick the best stream again.

**Comment by jchhenderson** [comment]
@skovatch - thanks for the tip - I have always had it unchecked.  ...but right now, unchecked means no Dr. Who downloads - will there be an update shortly that overcomes it?

...once again, thanks for your awesome and friendly support ion this 👍

**Comment by skovatch** [comment]
Hmm... I'll have to investigate a bit more when I'm not at work. I will push an update soon that fixes the metadata fetch when you choose 'Get Current Webpage', but you should still be able to download the show even though you see the "none"/"Not Available" in the queue.

**Comment by jchhenderson** [comment]
Yes, I can download with the "none/Not Available" shows. ...providng I have checked the "ignore dash" box as well.

**Comment by darfielddx** [comment]
I have followed advice. I have Prefer 25 FPS streams if available and Get
Higher Quality audio checked and the Force get_iplayer etc un checked.

I opened the iPlayer website and chose my program and still it will not
download. The message is below.

Any further advice. I am now u sing 1.13.6

Downloading Show 1/1:

BBC Download (ID=b0b624f9): Downloading A Very English Scandal: Series 1 -
Episode 3
INFO: 1 total programmes
WARNING: A UK TV licence is required to access BBC iPlayer TV content
legally
INFO: Downloading tv: 'A Very English Scandal: Series 1 - 3. Episode 3
(b0b624f9) [original]'
INFO: Downloaded: 0.00 MB (00:00:00) @ 0.00 Mb/s (hvfxsd1/bi) [audio+video]
WARNING: Unexpected size for file segment [1]
WARNING: Expected: 1769080  Downloaded: 785363
WARNING: Retrying download
WARNING: Retrying tv: 'A Very English Scandal: Series 1 - 3. Episode 3
(b0b624f9) [original]'
BBC Download (ID=b0b624f9): Download Cancelled
